plugin-dell/dphandler/managers.go

```diff
@@ -168,3 +170,91 @@ func getInfoFromDevice(uri string, deviceDetails dpmodel.Device, ctx iris.Contex
 	ctx.StatusCode(resp.StatusCode)
 	ctx.Write([]byte(respData))
 }
+
+//VirtualMediaActions performs insert and eject virtual media operations on the device based on the request
+func VirtualMediaActions(ctx iris.Context) {
+	uri := ctx.Request().RequestURI
+	uri = replaceURI(uri)
+	var deviceDetails dpmodel.Device
+	//Get device details from request
+	err := ctx.ReadJSON(&deviceDetails)
+	if err != nil {
+		log.Error("While trying to collect data from request, got: " + err.Error())
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString("Error: bad request.")
+		return
+	}
+	device := &dputilities.RedfishDevice{
+		Host:     deviceDetails.Host,
+		Username: deviceDetails.Username,
+		Password: string(deviceDetails.Password),
+		PostBody: deviceDetails.PostBody,
+	}
+
+	// insert virtual media request modifications
+	if strings.Contains(uri, "VirtualMedia.InsertMedia") {
+		var reqBody dpmodel.VirtualMediaInsert
+		err = json.Unmarshal(device.PostBody, &reqBody)
+		if err != nil {
+			errMsg := "While unmarshaling the request to insert a virtual media, got: " + err.Error()
+			log.Error(errMsg)
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(errMsg)
+			return
+		}
+
+		device.PostBody, err = json.Marshal(reqBody)
+		if err != nil {
+			log.Error(err.Error())
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(err.Error())
+			return
+		}
+	} else if strings.Contains(uri, "VirtualMedia.EjectMedia") {
+		payload := map[string]interface{}{}
+		device.PostBody, err = json.Marshal(payload)
+		if err != nil {
+			log.Error(err.Error())
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(err.Error())
+			return
+		}
+	}
+
+	statusCode, _, body, err := queryDevice(uri, device, http.MethodPost)
+	if err != nil {
+		errMsg := "while performing actions on virtual media, got: " + err.Error()
+		log.Error(errMsg)
+		ctx.StatusCode(statusCode)
+		ctx.WriteString(errMsg)
+		return
+	}
+
+	if statusCode == http.StatusNoContent {
+		log.Info("VirtualMediaActions is successful for URI ")
+		statusCode = http.StatusOK
+		body = vmActionsResponse()
+	}
+
+	ctx.StatusCode(statusCode)
+	ctx.Write(body)
+}
+
+// vmActionsResponse response is used for creating a final response for virtual media actions
+func vmActionsResponse() []byte {
```

Comment:
This is an ambiguous name rename to createVirtMediaActionResponse

Reply:
Addressed comments